store/sql_user_store.go

```diff
@@ -47,8 +47,26 @@ func NewSqlUserStore(sqlStore *SqlStore) UserStore {
 }
 
 func (us SqlUserStore) UpgradeSchemaIfNeeded() {
-	// ADDED for 2.0 REMOVE for 2.4
-	us.CreateColumnIfNotExists("Users", "Locale", "varchar(5)", "character varying(5)", model.DEFAULT_LOCALE)
+	// ADDED for 3.2 remove for 3.6
+	var data []*model.User
+	shouldUpdate := false
+	if _, err := us.GetReplica().Select(&data, "SELECT * FROM Users WHERE ThemeProps LIKE '%solaris%'"); err == nil {
+		for _, user := range data {
+			if user.ThemeProps["codetheme"] == "solarized_dark" {
+				user.ThemeProps["codetheme"] = "solarized-dark"
+				shouldUpdate = true
+			} else if user.ThemeProps["codetheme"] == "solarized_light" {
+				user.ThemeProps["codetheme"] = "solarized-light"
+				shouldUpdate = true
+			}
+
+			if shouldUpdate {
```

Comment:
You never set `shouldUpdate` back to `false`, so once you hit one user that has the bad theme it's going to update the rest of the users